### PR TITLE
Provide more CSS vars for customization purposes

### PR DESCRIPTION
**Pull request by axelboc**
Part of #555

### CODE REVIEW

File: src/styles/vars.css [hunk 1]
```diff
@@ -37,11 +37,16 @@
...
   --h5w-line--curveColor: var(--secondary-dark);
   --h5w-line--auxColor: var(--secondary);
+  --h5w-matrix--fontFamily: var(--monospace);
+  --h5w-matrix--cellBgAlt: var(--secondary-light-bg);
+  --h5w-matrix--indexCellBg: var(--primary-bg);
+  --h5w-matrix--indexCellBgAlt: var(--primary-light-bg);
+  --h5w-matrix--anchorCellBg: var(--primary-dark-bg);
+  --h5w-scalar-fontFamily: var(--monospace);
+  --h5w-raw-fontFamily: var(--monospace);
 }
```
Comment (axelboc) [inline src/styles/vars.css:52]: To see what it looks like with the default values, you can pull the branch and comment out this block.

File: src/styles/base.css [hunk 2]
```diff
@@ -39,6 +39,7 @@ dl {
   margin-bottom: 1rem;
 }
 
+code,
 pre {
   font-family: var(--monospace);
 }
```
Comment (axelboc) [inline src/styles/base.css:45]: So text wrapped in back ticks in storybook gets a nicer font family.

File: src/stories/Customization.stories.mdx [hunk 3]
```diff
@@ -48,9 +48,79 @@ as you see fit. For instance, if you'd like to change the color of the curve of
...
+#### Heatmap
+
+| Name                                    | Default                                      | Description                               |
+| --------------------------------------- | -------------------------------------------- | ----------------------------------------- |
+| `--h5w-colorBar-ticks--color`           | `var(--h5w-ticks--color, gray)`              | Color of color bar ticks                  |
+| `--h5w-colorBar-tickLabels--color`      | `var(--h5w-tickLabels--color, #333)`         | Text color of color bar tick labels       |
+| `--h5w-colorBar-tickLabels--fontFamily` | `var(--h5w-tickLabels--fontFamily, inherit)` | Font family of color bar tick labels      |
+| `--h5w-colorBar-tickLabels--fontSize`   | `0.75em`                                     | Font size of color bar tick labels        |
+| `--h5w-colorBar-bounds--color`          | `#333`                                       | Font size of color bar min and max bounds |
+| `--h5w-colorBar-bounds--fontSize`       | `0.875em`                                    | Font size of color bar min and max bounds |
```
Comment (axelboc) [inline src/stories/Customization.stories.mdx:94]: Seemed too long to have `--h5w-heatmap--colorBarTickLabelsFontFamily` 😂

File: src/stories/Customization.stories.mdx [hunk 4]
```diff
@@ -48,9 +48,79 @@ as you see fit. For instance, if you'd like to change the color of the curve of
...
+| `--h5w-matrix--cellBg`              | `transparent`                              | Background colour of cells (1/2)       |
+| `--h5w-matrix--cellBgAlt`           | `#eee`                                     | Background colour of cells (2/2)       |
+| `--h5w-matrix--indexCellBg`         | `#eee`                                     | Background colour of index cells (1/2) |
+| `--h5w-matrix--indexCellBgAlt`      | `lightgray`                                | Background colour of index cells (2/2) |
+| `--h5w-matrix--indexCellColor`      | `var(--h5w-matrix--color, inherit)`        | Text color of index cells              |
+| `--h5w-matrix--indexCellFontFamily` | `var(--h5w-matrix--fontFamily, monospace)` | Font family of index cells             |
+| `--h5w-matrix--indexCellFontSize`   | `var(--h5w-matrix--fontSize, inherit)`     | Font size of index cells               |
+| `--h5w-matrix--indexCellFontWeight` | `600`                                      | Font weight of index cells             |
+| `--h5w-matrix--anchorCellBg`        | `silver`                                   | Background colour of anchor cell       |
```
Comment (axelboc) [inline src/stories/Customization.stories.mdx:111]: I wasn't sure if I should go with `--h5w-matrix-indexCell--fontFamily` like for the Heatmap's color bar. What do you think?

File: src/stories/Customization.stories.mdx [hunk 5]
```diff
@@ -48,9 +48,79 @@ as you see fit. For instance, if you'd like to change the color of the curve of
...
+| `--h5w-axisLabels--fontWeight` | `600`         | Font weight of axis labels  |
+| `--h5w-grid--color`            | `gray`        | Color of grid lines         |
+| `--h5w-grid--opacity`          | `0.33`        | Opacity of grid lines       |
+| `--h5w-tooltip--color`         | `inherit`     | Text color of tooltip       |
+| `--h5w-tooltip--bgColor`       | `whitesmoke`  | Background color of tooltip |
+| `--h5w-tooltip--fontFamily`    | `inherit`     | Font family of tooltip      |
+| `--h5w-tooltip--fontSize`      | `0.75em`      | Font size of tooltip        |
+| `--h5w-tooltip--guideColor`    | `gray`        | Color of tooltip guide      |
+| `--h5w-tooltip--guideWidth`    | `1.5`         | Width of tooltip guide      |
+| `--h5w-tooltip--guideOpacity`  | `0.5`         | Opacity of tooltip guide    |
```
Comment (axelboc) [inline src/stories/Customization.stories.mdx:76]: I chose not to add a `-common-` prefix or similar to remain concise...

File: src/h5web/vis-packs/core/shared/TooltipMesh.tsx [hunk 6]
```diff
@@ -86,6 +86,8 @@ function TooltipMesh(props: Props) {
               className={styles.tooltip}
               top={tooltipTop}
               left={tooltipLeft}
+              unstyled
+              applyPositionStyle
```
Comment (axelboc) [inline src/h5web/vis-packs/core/shared/TooltipMesh.tsx:90]: To remove `!important` from the CSS.

**Review by axelboc — COMMENTED** [review]
Note that I chose to use some of the new default values, which I thought looked better. Notably, the color of tick and axis labels is now `#333` instead of `darkslategray`, and I used a very subtle gray background colour for the tooltip (`whitesmoke` instead of `--primary-light-bg`).

File: src/h5web/vis-packs/core/heatmap/ColorBar.module.css [hunk 7]
```diff
@@ -1,8 +1,6 @@
 .colorBar {
   display: grid;
   padding: 1rem;
-  color: var(--axis-primary-color);
```
Comment (loichuder) [inline src/h5web/vis-packs/core/heatmap/ColorBar.module.css:4]: This is replaced by the `color` in `.tick text` ?

File: src/h5web/vis-packs/core/shared/AxisSystem.module.css [hunk 8]
```diff
@@ -30,20 +29,22 @@
 }
 
 .tick > line {
-  stroke: var(--axis-secondary-color);
+  stroke: var(--h5w-ticks--colorColor, gray);
```
Comment (loichuder) [inline src/h5web/vis-packs/core/shared/AxisSystem.module.css:32]: `colorColor` ? :thinking:

File: src/stories/Customization.stories.mdx [hunk 9]
```diff
@@ -48,9 +48,80 @@ as you see fit. For instance, if you'd like to change the color of the curve of
...
+
+| Name                                | Default                                    | Description                            |
+| ----------------------------------- | ------------------------------------------ | -------------------------------------- |
+| `--h5w-matrix--color`               | `inherit`                                  | Text color of matrix vis               |
+| `--h5w-matrix--fontFamily`          | `monospace`                                | Font family of matrix vis              |
+| `--h5w-matrix--fontSize`            | `inherit`                                  | Font size of matrix vis                |
+| `--h5w-matrix--cellBg`              | `transparent`                              | Background colour of cells (1/2)       |
+| `--h5w-matrix--cellBgAlt`           | `#eee`                                     | Background colour of cells (2/2)       |
+| `--h5w-matrix--indexCellBg`         | `#eee`                                     | Background colour of index cells (1/2) |
+| `--h5w-matrix--indexCellBgAlt`      | `lightgray`                                | Background colour of index cells (2/2) |
```
Comment (loichuder) [inline src/stories/Customization.stories.mdx:106]: Perhaps that could be clearer by explaining it is for even/odd cells ? :shrug:

File: src/stories/Customization.stories.mdx (hunk 4 above)
Comment (loichuder) [reply src/stories/Customization.stories.mdx:111]: Yeah `--h5w-matrix-indexCell--fontFamily` pleases me more.

You could then have `--h5w-tooltip-guide--color` as well.

**Review by loichuder — APPROVED** [review]
Props to you :clap: 

I start to see the limitation of our approach tho: each and every CSS property needs to be put into variables which could lead to tremendous work. Let's roll with it for now and see how it goes.

File: src/h5web/vis-packs/core/heatmap/ColorBar.module.css (hunk 7 above)
Comment (axelboc) [reply src/h5web/vis-packs/core/heatmap/ColorBar.module.css:4]: yep

File: src/h5web/vis-packs/core/shared/AxisSystem.module.css (hunk 8 above)
Comment (axelboc) [reply src/h5web/vis-packs/core/shared/AxisSystem.module.css:32]: ha, last minute renaming gone wrong

**Pull request merged**